nucypher/blockchain/eth/actors.py

```diff
@@ -449,9 +474,25 @@ def divide_stake(self,
             raise self.MinerError(f"Cannot divide stake; Value ({target_value}) must be less "
                                   f"than the existing stake value {stake.value}.")
 
+        current_stake = self.stakes[stake_index]
+
+        new_stake_1 = Stake(owner_address=self.checksum_public_address,
+                            index=len(self.stakes)+1,
+                            start_period=current_stake.start_period,
+                            end_period=current_stake.end_period + additional_periods,
+                            value=target_value,
+                            economics=self.economics)
+
+        new_stake_2 = Stake(owner_address=self.checksum_public_address,
+                            index=len(self.stakes)+1,
+                            start_period=current_stake.start_period,
+                            end_period=current_stake.end_period + additional_periods,
+                            value=stake.value - target_value,
+                            economics=self.economics)
+
         # Ensure both halves are for valid amounts
-        validate_stake_amount(amount=target_value)
-        validate_stake_amount(amount=stake.value - target_value)
+        new_stake_1.validate_value()
+        new_stake_2.validate_value()
```

Comment:
This stakes are confused me because indecies and `new_stake_2.end_period` are wrong/unused. I hadn't immediately noticed  that they are here only for "value" checking.

Reply:
Yeah - its worth considering that perhaps we want `Stake` to play a larger role, taking on some of the contract staking methods.

Reply:
Following up: <https://github.com/nucypher/nucypher/issues/932>